client/src/sections/resources/IndustrialFireExtinguisher.tsx

```diff
@@ -0,0 +1,97 @@
+import Line from "../../components/Line";
+import Tile from "../../components/Tile";
+import useHave from "../../hooks/useHave";
+import useGet from "../../hooks/useGet";
+import { Step, useQuestStep } from "../../hooks/useQuest";
+import { $item } from "../../util/makeValue";
+
+const IndustrialFireExtinguisher = () => {
+    const foam = useGet("_fireExtinguisherCharge");
+
+    const knobUnfinished = useQuestStep("questL05Goblin") < Step.FINISHED;
+    const knobUnused = !useGet("fireExtinguisherHaremUsed");
+    const hasPants = useHave($item`Knob Goblin harem pants`);
+    const hasVeil = useHave($item`Knob Goblin harem veil`);
+    const canVisitTheThrone = hasPants && hasVeil;
+    const showKnob =
+        useQuestStep("questL05Goblin") > Step.UNSTARTED &&
+        !canVisitTheThrone &&
+        knobUnused &&
+        knobUnfinished;
+
+    const bridgeUnfinished = useQuestStep("questL09Topping") < 1;
+    const blechUnused = !useGet("fireExtinguisherChasmUsed");
+    const showBlech =
+        useQuestStep("questL09Topping") > Step.UNSTARTED &&
+        bridgeUnfinished &&
+        blechUnused;
+
+    const batwallsUnfinished = useQuestStep("questL04Bat") < 3;
+    const batUnused = !useGet("fireExtinguisherBatHoleUsed");
+    const showBat =
+        useQuestStep("questL04Bat") > Step.UNSTARTED &&
+        batwallsUnfinished &&
+        batUnused;
+
+    const bonerUnready = [
+        useGet("cyrptAlcoveEvilness"),
+        useGet("cyrptCrannyEvilness"),
+        useGet("cyrptNicheEvilness"),
+        useGet("cyrptNookEvilness"),
+    ].some((evilness) => evilness > 25);
+    const cyrptUnused = !useGet("fireExtinguisherCyrptUsed");
+    const showCyrpt =
+        useQuestStep("questL07Cyrptic") > Step.UNSTARTED &&
+        bonerUnready &&
+        cyrptUnused;
+
+    const desertIncomplete = useGet("desertExploration") < 100;
+    const desertUnused = !useGet("fireExtinguisherDesertUsed");
+    const showDesert =
+        useQuestStep("questL11Desert") > Step.UNSTARTED &&
+        desertIncomplete &&
+        desertUnused;
+
+    if (!useHave($item`industrial fire extinguisher`) || foam <= 0)
+        return <></>;
+
+    return (
+        <Tile
+            header='Industrial Fire Extinguisher'
+            imageUrl='/images/itemimages/exting2.gif'
+        >
+            <Line>
+                Remaining foam: {foam}. That's a total of {Math.min(foam / 10)}{" "}
+                polar vortices.
+            </Line>
+            {showBat && (
+                <Line>
+                    You can use Constricted Blast to blown down a wall in the
```

Comment:
this is a general comment: there's a ton of cruft on these sentences. i would start with one intro line and then add point-by-point quick hit summaries. here is my take:
```
Or, extinguish an ascension relevant zone:
• <b>Constricted Blast</b> -- Open a Bat Hole chamber
• <b>Foam Up the Place</b> -- Obtain the Knob Harem Outfit
• <b>Replace the Chill</b> -- Remove 10 evil from one Cyrpt zone
• <b>Cool it Down</b> -- Get 66% closer to a Blech House NC
• <b>Take a Drink</b> -- Attain 15 turns of Ultrahydrated
```
each of these should have a href for their relevant container zone -- don't link to the adv.php but the container is good so they can immediately go there. also, IMO, this should all be grayed out when unequipped and black when equipped, but that we can wait on better formatting for because doing conditional formatting on each line sounds like absolute hell